packages/rich-text/src/create.js

```diff
@@ -391,7 +406,7 @@ function createFromMultilineElement( {
 		} );
 
 		// Multiline value text should be separated by a double line break.
-		if ( index !== 0 ) {
+		if ( index !== 0 || startSeparator === true ) {
```

Comment:
Perhaps `shouldPrependSeparator` would be a better name? `should` signals a boolean, and with _prepend_ it seems clearer to me.